deploy/kubernetes/base/images/image.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: builtin
+kind: ImageTagTransformer
```

Comment:
What is the difference between using this and the `images` field that's in the kustomization?

Reply:
Before, we have use append these images with some base. For example, in previous structure, we list most of images in stable version. And stable version needs a base, which point to the base that has both linux and windows setup. Now if all other versions e.g., alpha, dev need to base on stable, all of them must share the same base setup as stable (which contains both linux and windows.). If for alpha, we only want to have linux as base instead of using stable as a base, we have to copy the same image list to the alpha version. 
Now with this ImageTagTransformer, we can do any combination easily. The image is just like a resource to use. We can allow alpha to use mixed setup as base, plus images, while stable to use only linux setup, plus images.

Reply:
> And stable version needs a base, which point to the base that has both linux and windows setup.

Isn't it the opposite? Stable has just linux, and alpha has linux + windows? So everything can still continue to be off of stable, and then override the images. Unless the issue is you can't have 1 kustomization.yaml with `images` that uses another kustomization.yaml that also has `images` as its base?

Reply:
I think I used the wrong example. My main point is this way we can have arbitrary combination  of os + image as needed. This will give the flexibility in case we do want e.g., windows+image for security setup reason (we don't want to have linux security set up mix with it). And putting all common things in the same place seems clean.

Reply:
The tags in this file are only for the "stable" overlay. If we want to follow this model, should we have a file for the images in every overlay?